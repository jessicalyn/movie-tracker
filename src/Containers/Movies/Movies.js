import React from 'react'
import { Component } from 'react'
import  Card  from '../../Components/Card/Card'
import { connect } from 'react-redux'
import './Movies.css'

export class Movies extends Component {

  render(){
    const { movies } = this.props
    const movieCards = movies.map(movie => {
      return <Card key={movie.id} movie={movie} />
    })
    return(
      <section className="movie-container">
        { movieCards }
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  movies: state.movies
})

export default connect(mapStateToProps)(Movies)
