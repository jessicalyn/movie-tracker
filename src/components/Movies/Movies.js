import React from 'react'
import { Component } from 'react'
import { Card } from '../Card/Card'

import './Movies.css'

export default class Movies extends Component {
  constructor(props) {
    super(props)
  }

  render(){
    const { movies } = this.props
    const movieCards = movies.map(movie => {
      return <Card key={movie.id} movie={movie} />
    })
    return(
      <section className="movie-container">
        { movieCards }
      </section>
    )
  }
}

