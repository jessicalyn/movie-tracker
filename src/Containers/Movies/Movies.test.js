import React from 'react'
import { Movies, mapStateToProps } from './Movies'
import { shallow } from 'enzyme'

describe('Movies', () => {
  describe('Movies', () => {
    let wrapper

    beforeEach(() => {
      const mockMovies = [
        { id: 1, title: "Us" },
        { id: 2, title: "Cars" },
        { id: 3, title: "A League of their Own" }
      ]
      wrapper = shallow(<Movies movies={ mockMovies }/>)
    })

    it('should match the snapshot with all data passed in', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
  
  describe('mapStateToProps', () => {
    it('should return an object with a movies array', () => {
      const mockState = {
        movies: [
          {title: "Cars", poster_path: "car.jpg"},
          {title: "How To Train Your Dragon", poster_path: "dragon.jpg"}
        ],
        user: {
          id: 1,
          name: "Taylor"
        }
      }
      const expected = {
        movies: [
          {title: "Cars", poster_path: "car.jpg"},
          {title: "How To Train Your Dragon", poster_path: "dragon.jpg"}
        ]
      }
      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })
})