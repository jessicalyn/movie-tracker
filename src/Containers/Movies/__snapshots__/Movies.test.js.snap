// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Movies should match the snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movies
    movies={
      Array [
        Object {
          "id": 1,
          "title": "Us",
        },
        Object {
          "id": 2,
          "title": "Cars",
        },
        Object {
          "id": 3,
          "title": "A League of their Own",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Card)
          movie={
            Object {
              "id": 1,
              "title": "Us",
            }
          }
        />,
        <Connect(Card)
          movie={
            Object {
              "id": 2,
              "title": "Cars",
            }
          }
        />,
        <Connect(Card)
          movie={
            Object {
              "id": 3,
              "title": "A League of their Own",
            }
          }
        />,
      ],
      "className": "movie-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "movie": Object {
            "id": 1,
            "title": "Us",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "movie": Object {
            "id": 2,
            "title": "Cars",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "movie": Object {
            "id": 3,
            "title": "A League of their Own",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Card)
            movie={
              Object {
                "id": 1,
                "title": "Us",
              }
            }
          />,
          <Connect(Card)
            movie={
              Object {
                "id": 2,
                "title": "Cars",
              }
            }
          />,
          <Connect(Card)
            movie={
              Object {
                "id": 3,
                "title": "A League of their Own",
              }
            }
          />,
        ],
        "className": "movie-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "movie": Object {
              "id": 1,
              "title": "Us",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "movie": Object {
              "id": 2,
              "title": "Cars",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "movie": Object {
              "id": 3,
              "title": "A League of their Own",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
